File: 2696 minimum-string-length-after-removing-substrings/solution.cpp
class Solution {
public:
    int minLength(string s) {
        vector<char> st(s.size());
        int top = -1;
        for(char c:s) {
            if(top!=-1 && ((c=='B' && st[top]=='A') || (c=='D' && st[top]=='C'))) top--;
            else st[++top] = c;
        }
        return top+1;
    }
};

File: 2696 minimum-string-length-after-removing-substrings/solution.go
func minLength(s string) int {
        st:=make([]rune,len(s))
        top:=-1
        for _,c:=range s {
            if top!=-1 && ((c=='B' && st[top]=='A') || (c=='D' && st[top]=='C')) {
                 top--
            } else {
                top++
                st[top] = c
            }
        }
        return top+1
}
